doc/dualtor/ICMP_Hardware_Offload_and_Protecion.md

```diff
@@ -0,0 +1,289 @@
+  * default:default:5000:NORMAL
+  * default:default:6000:RX
+  * default:Ethernet0:12000:NORMAL
+* Session Type:
+  * 'NORMAL': Install a normal icmp echo session that will transmit, receive icmp echo session packets and monitor the session.
+  * 'RX': Install a monitor only icmp echo session and set the tx_interval to zero which will not send any icmp echo session.
+* Interface Alias 'default' will set the SAI atttribute hw_lookup to true and NPU will perform forwarding lookup basd on dst_ip to determine the outgoing interface.
+
+#### MuxOrch
+This feature introduces a new config knob **switching_mode** to differentiate between normal software based switching and FRR protection switching that will use next hop protection group to switch traffic. MuxOrch will create next hop protection group for each mux port and maintain a mapping of mux port and next hop protection group based on this config knob.
+MuxOrch creates the IPinIP tunnel based on the peer_switch configuration. Currently IPinIP tunnel destination next hop is created when mux state changes to standby however with frr_protection switching_mode it will create the IPinIP tunnel destination next hop in advance and add this as the backup member of the next hop protection group.
+
+#### MuxCableOrch
+MuxCableOrch in orchagent is the component responsible for consuming mux state from App DB MUX_CABLE_TABLE and switching traffic. Currently this component updates all routes whenever a traffic switchover is needed. When **switching_mode** will be set to **frr-protection** in MuxOrch, MuxCableOrch will program the routes with the nexthop protection group as destination. With frr-protection mode in the event of traffic switching, SONiC will just toggle the members of nexthop protection group and will no longer need to reprogram all routes.
+
+Following diagram shows component level flow for traffic switching.
+<div align="center"> <img src=image/traffic_switching_flow.png width=700 /> </div>
+
+#### MuxNbrHandler
+Currently when FDB changes or neighbor changes, neighbors are updated based on the state of mux. With frr-protection neighbor destination will change based on toggling of protection group.
+
+### DB Schema Changes
+
+#### Config-DB
+Two new knobs in MUX_CABLE config table to support these features:
+  * **link_prober_type**
+    * software : create software based icmp echo session. This is default value.
+    * hardware : create hardware based icmp echo session.
+  * **switching_mode**
+    * normal                   : Use normal software for traffic switching. This is default value.
+    * frr-protection-switching : Use FRR protection group for traffic switching.
+
+```
+{
+    "MUX_CABLE": {
+        "Ethernet4": {
+	    "cable_type": "active-active",
+            "server_ipv4": "192.168.0.2/32",
+            "server_ipv6": "fc02:1000::30/128",
+            "soc_ipv4": "192.168.0.3/32",
+            "state": "auto",
+            "link_prober_type": "software/hardware",

```

Comment:
can this attribute `link_prober_type` be changed dynamically, or it is a read-once fixed value after boot up?